src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -420,5 +436,26 @@ public ImmutableList<string> GetApplicableTargetsForReference(ProjectInstance re
                 }
             }
         }
+
+        public bool RequiresTransitiveProjectReferences(ProjectInstance projectInstance)
+        {
+            // Outer builds do not get edges based on ProjectReference or their transitive closure, only inner builds do.
+            if (GetProjectType(projectInstance) == ProjectType.OuterBuild)
+            {
+                return false;
+            }
+
+            // special case for Quickbuild which updates msbuild binaries independent of props/targets. Remove this when all QB repos will have
+            // migrated to new enough Visual Studio versions whose Microsoft.Managed.After.Targets enable transitive references.
+            if (projectInstance.GetPropertyValue("UsingMicrosoftNETSdk").Equals("true", StringComparison.OrdinalIgnoreCase) &&
+                !projectInstance.GetPropertyValue("DisableTransitiveProjectReferences").Equals("true", StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
+            return projectInstance
+                       .GetPropertyValue(AddTransitiveProjectReferencesInStaticGraphPropertyName)
+                       .Equals("true", StringComparison.OrdinalIgnoreCase);
```

Comment:
Should this and all the other ones like it use `ConversionUtilities.ConvertStringToBool` instead? So evil people can put `!no` instead of `tRuE`?